### PR TITLE
Restore previous tests order

### PR DESCRIPTION
**Pull request by mjauvin**

### CODE REVIEW

**Comment by mjauvin** [comment]
I couldn't reproduce the errors while calling $model->{'relation'} for all relation types with REAL models... I am going to assume the test models are missing something and merge this fix.

**Comment by LukeTowers** [comment]
What is that conditional supposed to be checking for?

**Comment by mjauvin** [comment]
> What is that conditional supposed to be checking for?

Not entirely sure, to be honest.

**Comment by LukeTowers** [comment]
> Not entirely sure, to be honest.

I don't really like that answer 😜 time to go digging through the history I guess 😂

File: src/Database/Model.php [hunk 1]
```diff
@@ -1026,6 +1022,10 @@ protected function performDeleteOnRelations()
                         continue;
                     }
 
+                    if (!$relation = $this->{$name}) {
```
Comment (LukeTowers) [inline src/Database/Model.php:1026]:
```suggestion
                    // Attempt to load the related record(s)
                    if (!$relation = $this->{$name}) {
```

File: src/Database/Traits/SoftDelete.php [hunk 2]
```diff
@@ -115,10 +115,10 @@ protected function performSoftDeleteOnRelations()
         $definitions = $this->getRelationDefinitions();
         foreach ($definitions as $type => $relations) {
             foreach ($relations as $name => $options) {
-                if (!$relation = $this->{$name}) {
+                if (!array_get($options, 'softDelete', false)) {
                     continue;
                 }
-                if (!array_get($options, 'softDelete', false)) {
+                if (!$relation = $this->{$name}) {
```
Comment (LukeTowers) [inline src/Database/Traits/SoftDelete.php:122]:
```suggestion
                // Attempt to load the related record(s)
                if (!$relation = $this->{$name}) {
```

**Pull request merged**